TomLonghurst.PullRequestScanner.AzureDevOps/Extensions/AzureDevOpsHttpClientExtensions.cs

```diff
@@ -0,0 +1,137 @@
+﻿using System.Runtime.CompilerServices;
+using TomLonghurst.PullRequestScanner.AzureDevOps.Http;
+using TomLonghurst.PullRequestScanner.AzureDevOps.Models;
+
+namespace TomLonghurst.PullRequestScanner.AzureDevOps.Extensions;
+
+internal static class AzureDevOpsHttpClientExtensions
+{
+    public static async IAsyncEnumerable<AzureDevOpsPullRequest> GetPullRequests(this AzureDevOpsHttpClient azureDevOpsHttpClient, string repositoryId, [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        var response = await azureDevOpsHttpClient.GetAll<AzureDevOpsPullRequestsResponse>(
+            $"_apis/git/pullrequests?searchCriteria.status=all&searchCriteria.includeLinks=false&searchCriteria.repositoryId={repositoryId}&includeCommits=true&api-version=7.1-preview.1", cancellationToken)
+            .ConfigureAwait(false);
+
+        foreach (var azureDevOpsPullRequestsResponse in response)
+        {
+            foreach (AzureDevOpsPullRequest azureDevOpsPullRequest in azureDevOpsPullRequestsResponse.PullRequests)
+            {
+                if (cancellationToken.IsCancellationRequested)
+                {
+                    yield break;
+                }
+
+                yield return azureDevOpsPullRequest;
+            }
+        }
+    }
+
+    public static async IAsyncEnumerable<AzureDevOpsPullRequestIteration> ListPullRequestIterations(this AzureDevOpsHttpClient azureDevOpsHttpClient, string repositoryId, int pullRequestId, [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        var response = await azureDevOpsHttpClient.Get<AzureDevOpsPullRequestIterationResponse>(
+            $"_apis/git/repositories/{repositoryId}/pullRequests/{pullRequestId}/statuses?api-version=7.1-preview.1", cancellationToken).ConfigureAwait(false);
+
+        if (response is null)
+        {
+            yield break;
+        }
+
+        foreach (var azureDevOpsPullRequestIteration in response.Value)
+        {
+            if (cancellationToken.IsCancellationRequested)
+            {
+                yield break;
+            }
+
+            yield return azureDevOpsPullRequestIteration;
```

Comment:
Converting this into an IAsyncEnumerable isn't really benefitting us here because we already have all of the items after the API call anyway.